.circleci/config.yml

```diff
@@ -0,0 +1,79 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/desktop/desktop
+    parallelism: 1
+    shell: /bin/bash --login
+    environment:
+      CIRCLE_ARTIFACTS: /tmp/circleci-artifacts
+      CIRCLE_TEST_REPORTS: /tmp/circleci-test-results
+    macos:
+      xcode: '9.3.0'
+    steps:
+    - checkout
+    - run: mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+    - restore_cache:
+        keys:
+        # when lock file changes, use increasingly general patterns to restore cache
+        - yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}
+        - yarn-packages-v1-{{ .Branch }}-
+        - yarn-packages-v1-
+    - run:
+        name: Install dependencies
+        command: yarn install --force
+    - save_cache:
+        key: yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}
+        paths:
+        - ~/.cache/yarn
+        - vendor/bundle
+        - .eslintcache
+        - ~/.electron
+        - ~/Library/Caches/Yarn/v1
+        - ./node_modules
```

Comment:
Going with [my comment above](https://github.com/desktop/desktop/pull/5272/files#r207064438), I think adding `./app/node_modules` would help by caching that folder as well.